### ISSUE TITLE
Only retrieve variables needed for trajectories in era5 domain data

### ISSUE THREAD

**Issue by leifdenby** [EUREC4A-UK/lagtraj#26]
Snippet from @sjboeing that we want to mimick

```python
     def interp(self, kwargs, **interp_to):
        """
        Implements xarray.interp by first slicing out the necessary data from
        individual era5 files, merging these and then interpolating
        """
        idx_padding = 1
        interp_dims = interp_to.keys()
        datasets_slices = []
        for ds in self.datasets.values():
            vars_for_traj=["u", "v", "sp", "z", "t", "q", "lsm"]
            vars_in_ds=list(set(ds.variables))
            if len(vars_in_ds) > 0:
                ds_slice=ds[vars_in_ds].sel(time=interp_to['time'],
                    lat=slice(
                        interp_to['lat'] + 0.2, interp_to['lat'] - 0.2
                    ),
                    lon=slice(
                        interp_to['lon'] - 0.2, interp_to['lon'] + 0.2,
                    ),)
                ds_slice.load()
                ds_i=ds_slice.interp(lat=interp_to['lat'],
                    lon=interp_to['lon'])
                ds_slice.close()
                datasets_slices.append(ds_i)

        ds_at_l=xr.merge(datasets_slices)
        ds_at_l.load()
        return ds_at_l

    def interp(self, kwargs, **interp_to):
        """
        Implements xarray.interp by first slicing out the necessary data from
        individual era5 files, merging these and then interpolating
        """
        idx_padding = 1
        interp_dims = interp_to.keys()
        datasets_slices = []
        for ds in self.datasets.values():
            vars_for_traj=["u", "v", "sp", "z", "t", "q", "lsm"]
            vars_in_ds=list(set(ds.variables))
            if len(vars_in_ds) > 0:
                ds_slice=ds[vars_in_ds].sel(time=interp_to['time'],
                    lat=slice(
                        interp_to['lat'] + 0.2, interp_to['lat'] - 0.2
                    ),
                    lon=slice(
                        interp_to['lon'] - 0.2, interp_to['lon'] + 0.2,
                    ),)
                ds_slice.load()
                ds_i=ds_slice.interp(lat=interp_to['lat'],
                    lon=interp_to['lon'])
                ds_slice.close()
                datasets_slices.append(ds_i)

        ds_at_l=xr.merge(datasets_slices)
        ds_at_l.load()
        return ds_at_l
```

**Comment by sjboeing** [comment]
Hi @leifdenby:

A quick update after some more testing:
- It appears that applying .sel on a dataset is faster than applying it on the individual arrays
- Calculating the slice longitude and latitude time indices using sel is just a bit slow (5% or so performance loss). I have come up with a temporary suggestion in the attachment.
- I think "load" is quicker than "compute" (which surprises me a bit)
- The snippet above misses the variable selection (I probably sent you a wrong file)

[slightly_less_ugly_load.txt](https://github.com/EUREC4A-UK/lagtraj/files/4961497/slightly_less_ugly_load.txt)

**Comment by sjboeing** [comment]
@leifdenby: did a little more work on this. Below is a suggestion for interp that seems to have reasonable performance (it takes a "variables" argument, sel could be modelled on it)

```python
    def interp(self, kwargs, variables=None, **interp_to):
        """
        Implements xarray.interp by first slicing out the necessary data from
        individual era5 files, merging these and then interpolating
        """
        interp_dims = interp_to.keys()
        datasets_slices = []
        for ds in self.datasets.values():
            if variables==None:
                vars_to_get=ds.variables
            else:
                vars_to_get=list(set(ds.variables) & set(variables))
            if len(vars_to_get)==0:
                continue
            slices={}
            interp_other_than_exact_time={}
            for d in set(ds.dims).intersection(interp_dims):
                if d=='time':
                    # Only select one time slice if there is an exact match
                    if interp_to['time'].values in ds['time'].values:
                        slices[d]=interp_to[d]
                        continue
                interp_other_than_exact_time[d]=interp_to[d]
                # interp_to can hold data arrays or scalars
                if(type(interp_to[d])==xr.core.dataarray.DataArray):
                    d_interp_val=interp_to[d].values
                else:
                    d_interp_val=interp_to[d]
                d_vals_array=ds[d].values
                d_vals_smaller=d_vals_array[d_vals_array<d_interp_val]
                d_vals_greater=d_vals_array[d_vals_array>d_interp_val]
                assert len(d_vals_smaller)>0
                assert len(d_vals_greater)>0
                d_slice_min=np.nanmax(d_vals_smaller)
                d_slice_max=np.nanmin(d_vals_greater)
                # Slice direction depends on how variables are ordered
                if d_vals_array[0]<d_vals_array[1]:
                     slices[d]=slice(d_slice_min,d_slice_max)
                else:
                     slices[d]=slice(d_slice_max,d_slice_min)
            da_v_slice = ds[vars_to_get].sel(**slices)
            da_v_slice.load()
            datasets_slices.append(da_v_slice)
            da_v_slice.close()
        ds_slice = xr.merge(datasets_slices, compat="override")
        return ds_slice.interp(**interp_other_than_exact_time, kwargs=kwargs)
```

**Comment by sjboeing** [comment]
Implemented in #34 and #35.
